tensorboard/webapp/metrics/views/right_pane/settings_view_component.ts

```diff
@@ -187,6 +187,11 @@ export class SettingsViewComponent {
     lowerValue: number;
     upperValue: number;
   }) {
+    // When range values are updated but useRange is still false, this means the
+    // current selection state is single. We flip it to range selection.
+    if (!this.useRangeSelectTime) {
+      this.useRangeSelectTimeToggled.emit();
```

Comment:
Invoking the spirit of Stephan: You are introducing composite actions (two actions fired in quick succession asking for two different states to be mutated) which is a bit of a code smell that we want to avoid. 

I haven't had enough time to think about it but one idea that comes to mind is to move this logic (or some equivalent) the reducer layer.